### PR TITLE
docs: simplify source map docs

### PR DESCRIPTION
**Pull request by bmorelli25**
~NOTE: DO NOT MERGE YET. `elasticsearch-ci/docs` will fail until elastic/apm-server#3678 is merged.~ OK to merge now.

---

This PR simplifies the source map documentation in the RUM Reference. It moves the step by step content to the APM Server's "How-to guides" section, so we can have an end-to-end guide for applying source maps to error stack traces in one place.

For https://github.com/elastic/apm-server/issues/3564.
A component of https://github.com/elastic/apm-server/pull/3678.

### CODE REVIEW

File: docs/sourcemap.asciidoc [hunk 1]
```diff
@@ -1,147 +1,26 @@
...
-== Source Maps
+== Source maps
 
-TIP: Additional information on the source map endpoint is available in the
-APM Server's {apm-server-ref-v}/sourcemaps.html[source map documentation].
-Don't forget, you must enable {apm-server-ref-v}/configuration-rum.html[RUM support] in the APM Server for this endpoint to work.
+Minifying JavaScript bundles is a common practice in production as it can improve the load time and network latency of your application.
+However, minified code by itself can be hard to debug.
+For this reason, Elastic APM supports source maps.
+A source map maps minified files back to the original source code,
```
Comment (vigneshshanmugam) [inline docs/sourcemap.asciidoc:7]:
```suggestion
A source map is a file that maps minified files back to the original source code,
```

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
I think I found the problem. Fix is in https://github.com/elastic/apm-server/pull/3712.

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by bmorelli25** [comment]
Finally!

**Pull request merged**